src/Autocomplete/src/AutocompleteResults.php

```diff
@@ -14,11 +14,25 @@
 final class AutocompleteResults
 {
     /**
-     * @param list<array{label: string, value: mixed}> $results
+     * @param list<array{text: string, value: mixed, group_by?: string}> $options
+     * @param list<array{label: string, value: string}>                  $optgroups
      */
     public function __construct(
-        public array $results,
+        public array $options,
+        public array $optgroups,
         public bool $hasNextPage,
     ) {
     }
+
+    /**
+     * @return array{options: array{text: string, value: mixed, group_by?: string}, optgroups: array{label: string, value: string}}
+     */
+    public function getResults(): array
+    {
+        if (!empty($this->optgroups)) {
```

Comment:
```suggestion
        if ($this->optgroups) {
```
Based on Symfony CS :)